### ISSUE TITLE
cudnn ReLU now with zero memory

### ISSUE THREAD

**Issue by soumith** [eladhoffer/ImageNet-Training#1]
hey, just FYI, to save lots of memory, and make googlenet trainable within reasonable memory, I added an in-place mode to cudnn.ReLU. This cuts down memory usage of a network by almost half, as now the relu is not using separate states, but is just doing the operations in-place in the buffers of the above and below modules.

Check the README for more details:
https://github.com/soumith/cudnn.torch

in summary, you have to use: cudnn.ReLU(true) -- for in-place

hope it helps.
--s

**Comment by eladhoffer** [comment]
Thanks! I'll try that. Does it hurt performance?

**Comment by soumith** [comment]
the speed is the same, it does not hurt performance.
